pynest/nest/lib/hl_api_types.py

```diff
@@ -442,16 +450,27 @@ def index(self, node_id):
             If the node ID is not in the `NodeCollection`.
         """
         index = sli_func('Find', self._datum, node_id)
+
         if index == -1:
             raise ValueError('{} is not in NodeCollection'.format(node_id))
+
         return index
 
+    def __array__(self, dtype=None):
+        ''' Numpy compatibility function '''
+        return numpy.array(self.tolist(), dtype=dtype)
+
     def __getattr__(self, attr):
         if attr == 'spatial':
             metadata = sli_func('GetMetadata', self._datum)
             val = metadata if metadata else None
             super().__setattr__(attr, val)
             return self.spatial
+
+        # numpy compatibility check
+        if attr.startswith('__array_'):
+            raise AttributeError
```

Comment:
Please add a comment on what this actually does.

Reply:
done